### PR TITLE
Update gradle and java versions to meet with latests IDE requrements

### PR DESCRIPTION
**Pull request by Gelassen**
I have updated gradle and java versions to meet with latest Android Studio requirements

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -12,4 +12,7 @@ android.useAndroidX=true
 # Automatically convert third-party libraries to use AndroidX
 android.enableJetifier=true
 # Kotlin code style for this project: "official" or "obsolete":
-kotlin.code.style=official
\ No newline at end of file
+kotlin.code.style=official
+android.defaults.buildfeatures.buildconfig=true
+android.nonTransitiveRClass=false
+android.nonFinalResIds=false
```
Comment (mhsmith) [inline gradle.properties]: Why are these necessary?

`nonTransitiveRClass` is in the default gradle.properties file generated by the Android Studio new project wizard, but with a value of `true`.

The other two are not in the default gradle.properties file at all.

File: app/build.gradle [hunk 2]
```diff
@@ -33,6 +33,7 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+    namespace 'com.chaquo.python.console'
```
Comment (mhsmith) [inline app/build.gradle]: This is an important setting, so it should appear at the top of the `android` block, which is where the Android Studio new project wizard would put it.

File: gradlew [hunk 3]
```diff
@@ -1,74 +1,129 @@
-#!/usr/bin/env bash
+#!/bin/sh
+
+#
+# Copyright © 2015-2021 the original authors.
```
Comment (mhsmith) [inline gradlew]: Where did these versions of gradlew and gradle-wrapper.jar come from? They're not the same as the ones generated by the Android Studio new project wizard, which apparently haven't changed since Arctic Fox.

File: gradle.properties (hunk 1 above)
Comment (Gelassen) [reply gradle.properties]: I have used Android Studio migration assistant, this two lines had been added automatically. I don't yet their impact on the project, it will require an extra research to figure out. 

If tests passed, in my opinion we can merge it into the master.

File: gradlew (hunk 3 above)
Comment (Gelassen) [reply gradlew]: I doubled checked, indeed - they are different. What I just did is run Android Studio migration assistant to update versions. My thought is during migration from version to version it might brought this changes.
 
My proposal would be the same: if tests passes, it is sage to merge it into the master.

File: app/build.gradle (hunk 2 above)
Comment (Gelassen) [reply app/build.gradle]: Done.

File: gradlew (hunk 3 above)
Comment (Gelassen) [reply gradlew]: I checked code on the emulator (API 33) and it compiles and works without noticed issues.

File: gradle.properties (hunk 1 above)
Comment (mhsmith) [reply gradle.properties]: I've reverted to the gradle.properties from the new project wizard, because that's more representative of what most users will have.

File: gradlew (hunk 3 above)
Comment (mhsmith) [reply gradlew]: Strange, I just tried the migration assistant myself, and it didn't change gradlew, gradlew.bat or gradle-wrapper.jar. So as above, I've reverted all three files to be more representative of what most users will have.

**Comment by mhsmith** [comment]
Thanks for the contribution, and sorry for the delay.
